### PR TITLE
Add a standard AttributeDescriptorFinder implementation in expr.

### PR DESCRIPTION
**Pull request by ozevren**

### CODE REVIEW

File: mixer/pkg/expr/finder.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import (
+	"fmt"
+	"sort"
+
+	configpb "istio.io/istio/mixer/pkg/config/proto"
+	"istio.io/istio/mixer/pkg/pool"
+)
+
+// EmptyFinder is a AttributeDescriptorFinder with no attributes
+var EmptyFinder AttributeDescriptorFinder = finder{
```
Comment (mandarjog) [inline mixer/pkg/expr/finder.go]: where is empty finder used?
even without allocating an zero length map, you should be able to get the same `GetAttribute()` behaviour.

**Comment by mandarjog** [comment]
Shouldn't this be defined in `pkg/attribute`, why is this in `expr` ?

File: mixer/pkg/expr/finder.go (hunk 1 above)
Comment (ozevren) [reply mixer/pkg/expr/finder.go]: In config declaration mainly. https://github.com/ozevren/istio/blob/runtime2-config/mixer/pkg/runtime2/config/snapshot.go This is the only place it is used. I'll apply the same implicit map pattern there as well and remove the explicit initializations.

Thanks!

**Comment by ozevren** [comment]
I placed this in expr, since AttributeDescriptorFinder is defined here.

**Pull request closed without merging**